test/integration/loans.test.ts

```diff
@@ -137,18 +162,420 @@ describe('Loans', () => {
+
+          
+          //simulate that excess profit was made by the user on the borrowed money 
+          const borrowedFundsEscrowAddress = await diamond.connect( borrower ).getLoanEscrow(loanId)
+          
+          const tTokenAddress = await diamond.connect( borrower ).getTTokenFor( lendingToken.address )
+          
+          console.log('tTokenAddress 1 ', tTokenAddress.toString())
+           let tTokenEscrowBalance = await lendingToken.balanceOf(tTokenAddress)
+           console.log('tTokenEscrowBalance 1 ', tTokenEscrowBalance.toString())
+
+           expect(tTokenEscrowBalance).to.equal(99700030000)
+
+
+          let loanEscrowBalance = await lendingToken.balanceOf(borrowedFundsEscrowAddress)
+          console.log('loanEscrowBalance 1 ', loanEscrowBalance.toString())
+
+          expect(loanEscrowBalance).to.equal(100000000)
+
+
+          console.log('borrower', borrower.getAddress() )
+
+          console.log('add funds to borrowed escrow ',borrowedFundsEscrowAddress, lendingToken.signer.getAddress()  ) 
+         // await lendingToken.transfer(borrowedFundsEscrowAddress,  30     )
+ 
+          
+
+          const borrowerAddress = await borrower.getAddress()
+          const borrowerBalance = await lendingToken.balanceOf(borrowerAddress) 
+
+
+          //simulating that a profit was made, multiplying the borrowed $$ escrow 
+          await lendingToken
+          .connect(ethers.provider.getSigner(borrowerAddress))
+          .transfer(borrowedFundsEscrowAddress, 30 * 10**(lendingTokenDecimals))
+
+          loanEscrowBalance = await lendingToken.balanceOf(borrowedFundsEscrowAddress)
+          console.log('loanEscrowBalance 2 ', loanEscrowBalance.toString())
+          expect(loanEscrowBalance).to.equal(130000000)
+
+          console.log('balanceOf', borrowerBalance.toString() )
+
+          console.log('market.lendingToken',market.lendingToken)
+
+
+          const balanceLeftToRepay =  await diamond.getTotalOwed( loanId )
+          console.log('balanceLeftToRepay',balanceLeftToRepay.toString())
+
+
+
+          
+
+          await lendingToken
+          .connect(ethers.provider.getSigner(borrowerAddress))
+          .approve(diamond.address, balanceLeftToRepay)
+
+         
+          await diamond.connect( borrower ).repayLoan(loanId, 100010000)
+
+          
+          const totalOwedAfterRepay = await diamond.getTotalOwed( loanId )
+
+
+          console.log('totalOwedAfterRepay',  totalOwedAfterRepay.toString()  )
+
+
+          const loanData = await diamond.getLoan(loanId)
+          console.log('loanData',  ) 
+
+         
+          console.log('balanceLeftToRepay 2',balanceLeftToRepay.toString())
+
+          console.log('balanceOf After', borrowerBalance.toString() )
+
+
+
+          tTokenEscrowBalance = await lendingToken.balanceOf(tTokenAddress)
+           console.log('tTokenEscrowBalance 2 ', tTokenEscrowBalance.toString())
+           expect(tTokenEscrowBalance).to.equal(99801540000)  //99700030000
```

Comment:
instead of hardcoding this value, because this value depends on how much was originally in the TToken, lets grab the TToken balance before and after the repay. The difference should be `the loan repayment + profit fee`